<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="karltayeb" />

<meta name="date" content="2021-04-26" />

<title>GSEA Method comparisons</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">logistic_susie_gsea</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GSEA Method comparisons</h1>
<h4 class="author">karltayeb</h4>
<h4 class="date">2021-04-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>logistic_susie_gsea/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210205code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210205code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210205)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatreee5b024ab45bc6bdb251d39f7f2f92e8f2cac614etargetblanke5b024aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karltayeb/logistic_susie_gsea/tree/e5b024ab45bc6bdb251d39f7f2f92e8f2cac614e" target="_blank">e5b024a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatreee5b024ab45bc6bdb251d39f7f2f92e8f2cac614etargetblanke5b024aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karltayeb/logistic_susie_gsea/tree/e5b024ab45bc6bdb251d39f7f2f92e8f2cac614e" target="_blank">e5b024a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/GO_simulation_cache/

Untracked files:
    Untracked:  GO_simulation_cache/
    Untracked:  _targets.R
    Untracked:  _targets/
    Untracked:  _targets2.R
    Untracked:  analysis/cache/
    Untracked:  analysis/initial_exploration.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  cache/
    Untracked:  code/dream.targets.R
    Untracked:  code/fit.R
    Untracked:  code/gseabenchmark.targets.R
    Untracked:  code/logistic_susie.R
    Untracked:  code/plots.R
    Untracked:  code/score.R
    Untracked:  code/simulations.R
    Untracked:  code/utils.R
    Untracked:  data/DREAM_modules/
    Untracked:  data/GO_Membership.RData
    Untracked:  data/WebGestalt/
    Untracked:  data/WebGestalt_cache/
    Untracked:  data/gene_list.txt
    Untracked:  output/WebGestalt/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   analysis/GO_simulation.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/gsea_comparisons.Rmd</code>) and HTML (<code>docs/gsea_comparisons.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/e5b024ab45bc6bdb251d39f7f2f92e8f2cac614e/analysis/gsea_comparisons.Rmd" target="_blank">e5b024a</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/gsea_comparisons.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic_susie_gsea/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/gsea_comparisons.html" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/8fbadf2bafc56576cd089128037d98855a5c49c0/analysis/gsea_comparisons.Rmd" target="_blank">8fbadf2</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(“analysis/gsea_comparisons.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>We try to replicate some analysis from othe GSEA papers/ GSEA Benchmark papers, and extend them to compare multiple GSEA methods under consideration.</p>
<div id="overview" class="section level2">
<h2>Overview</h2>
<pre class="r"><code>library(targets)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✓ ggplot2 3.3.3     ✓ purrr   0.3.4
✓ tibble  3.1.1     ✓ dplyr   1.0.5
✓ tidyr   1.1.3     ✓ stringr 1.4.0
✓ readr   1.4.0     ✓ forcats 0.5.1</code></pre>
<pre class="r"><code># load and row_bind all targets matching a pattern
tar_agg &lt;- function(pattern){
  load.env &lt;- new.env()
  tar_load(matches(pattern), envir = load.env)
  result &lt;- grep(pattern , names(load.env), value=TRUE)
  result &lt;- do.call(&quot;list&quot;, mget(result, envir = load.env))
  result &lt;- bind_rows(result)
}</code></pre>
</div>
<div id="gsea-benchmarkr" class="section level2">
<h2>GSEA BenchmarkR</h2>
<div id="relevance-scores" class="section level3">
<h3>Relevance scores</h3>
<p><code>GSEABenchmarkeR</code> is a package for benchmarking GSEA methods. They provide uniformly processed expression data from TCGA, and for each disease they provide “relevance scores” for gene sets in GO-BP, GO-MF, and KEGG. These relevance scores were derived from MalaCards. I’m not sure exactly how they were generated but they are supposed to serve as Silver standard to benchmark performance of GSEA methods.</p>
<ol style="list-style-type: decimal">
<li>Took the gene sets used in GSEABenchmarkeR paper [cite] from GO and KEGG (downloaded from <code>EnrichmentBrowser</code>, filter down to gene sets larger that 5 genes and smaller than 500 genes).</li>
<li>Performed a vanilla differential expression analysis from the <code>GSEABenchmarkeR</code> documentation for each TCGA expression set. We construct gene lists by taking all genes with Benjamini Hochberg corrected pvalue <span class="math inline">\(&lt; 0.05\)</span></li>
<li>Fit gene-list based GSEA methods to the resulting gene lists</li>
<li>Compute a “relevance score” which compares the ranking produced by the GSEA method with the relevance scores.</li>
</ol>
<p>Fishers exact test, SuSiE, and Logistic SuSiE have the highest relevance according to this relevance scores Ranks for SuSie, Logistic SuSiE, and mr.ash generated from PIPs, ranks for FET generated from BH corrected p-values.</p>
<pre class="r"><code>pattern &lt;- &#39;^gseabenchmark.score_([^_])*$&#39;
result &lt;- tar_agg(pattern)

tar_load(&#39;gseabenchmark.score_lasso&#39;)
tar_load(&#39;gseabenchmark.score_susie.veb.boost&#39;)
tar_load(&#39;gseabenchmark.score_logistic.susie.veb.boost&#39;)
tar_load(&#39;gseabenchmark.score_lasso&#39;)
tar_load(&#39;gseabenchmark.score_elastic.net&#39;)
tar_load(&#39;gseabenchmark.score_fet&#39;)

result &lt;- bind_rows(
  gseabenchmark.score_fet,
  gseabenchmark.score_lasso,
  gseabenchmark.score_susie.veb.boost,
  gseabenchmark.score_logistic.susie.veb.boost,
  gseabenchmark.score_lasso,
  gseabenchmark.score_elastic.net
)

result %&gt;% 
  mutate(precent_optimal_relevance = relevance / opt.relevance) %&gt;%
  ggplot(aes(x=method, y=precent_optimal_relevance, fill=method)) +
  geom_boxplot(position=&#39;dodge&#39;) + facet_wrap(vars(ID)) </code></pre>
<p><img src="figure/gsea_comparisons.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/unnamed-chunk-2-1.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>result %&gt;% 
  mutate(precent_optimal_relevance = relevance / opt.relevance) %&gt;%
  ggplot(aes(x=method, y=precent_optimal_relevance, fill=method)) +
  geom_col(position=&#39;dodge&#39;) + facet_wrap(vars(ID)) </code></pre>
<p><img src="figure/gsea_comparisons.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/unnamed-chunk-2-2.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>result %&gt;% 
  ggplot(aes(x=method, y=relevance, fill=method)) +
  geom_col(position=&#39;dodge&#39;) +
  facet_wrap(vars(ID)) + 
  theme(axis.text.x=element_blank(), axis.ticks.x=element_blank())</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-3">
Past versions of unnamed-chunk-2-3.png
</button>
</p>
<div id="fig-unnamed-chunk-2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/unnamed-chunk-2-3.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="number-of-reported-gene-sets" class="section level2">
<h2>Number of reported gene sets</h2>
<p>Although FET and SuSiE both have high relevance, both SuSiE methods return far fewer gene sets. Perhaps this is an apples and oranges comparison. For SuSiE and mr.ash we include gene sets with <span class="math inline">\(PIP &gt; 0.95\)</span>. For FET we include gene sets with a Benjamini Hochberg corrected p-value <span class="math inline">\(&lt; 0.05\)</span>. For the regularized regression methods we include any gene set with nonzero coefficient.</p>
<pre class="r"><code>pattern &lt;- &#39;^gseabenchmark.size_([^_])*$&#39;
result &lt;- tar_agg(pattern)

t &lt;- 0.95

result %&gt;% ungroup() %&gt;% 
  unnest(c(positives, thresh)) %&gt;%
  filter(near(thresh, t)) %&gt;%
  ggplot(aes(x=method, y=log2(positives), fill=method)) +
  geom_boxplot(position=&#39;dodge&#39;)</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/gseabenchmark_sizes-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gseabenchmark_sizes-1">
Past versions of gseabenchmark_sizes-1.png
</button>
</p>
<div id="fig-gseabenchmark_sizes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/gseabenchmark_sizes-1.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>result %&gt;% ungroup() %&gt;% 
  unnest(c(positives, thresh)) %&gt;%
  filter(near(thresh, t)) %&gt;% 
  mutate(log2pos = log2(positives)) %&gt;%
  pivot_wider(
    id_cols = ID, names_from=method, values_from = log2pos) %&gt;%
  GGally::ggpairs(columns = 2:7)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;GGally&#39;:
  method from   
  +.gg   ggplot2</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/gseabenchmark_sizes-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gseabenchmark_sizes-2">
Past versions of gseabenchmark_sizes-2.png
</button>
</p>
<div id="fig-gseabenchmark_sizes-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/gseabenchmark_sizes-2.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Across the board lasso and elastic net did did worse than Fisher’s exact test. That’s unexpected. All the bayesian variable selection methods select only a handful of components.</p>
<p>Logistic SuSiE and SuSiE both have favorable relevance scores, lathough not quite as good as fisher’s exact test.</p>
</div>
<div id="dream-string-consensus-modules" class="section level2">
<h2>DREAM STRING Consensus modules</h2>
<p>These are the gene sets used in the <code>gerr</code> paper. The Diseas Module Identification DREAM Challenge was a crowd-sourced attempt at detecting disease related gene modules. The <code>gerr</code> paper showed results using 377 gene modules as gene lists. We repeat that analysis here using a few different gene sets: GO-Biological Process, GO-BP noredundant from <code>WebGestaltR</code> and MSigDB C2 curated gene sets.</p>
<pre class="r"><code>pattern &lt;- &#39;^dream.modules.size_([^_])*$&#39;
result &lt;- tar_agg(pattern)

t &lt;- 0.95 

result %&gt;% ungroup() %&gt;%
  unnest(c(positives, thresh)) %&gt;%
  filter(near(thresh, t)) %&gt;%
  ggplot(aes(x=GS_ID, y=positives, fill=method)) +
  geom_boxplot(position=&#39;dodge&#39;) + scale_y_continuous(trans=&#39;log10&#39;) </code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Removed 1388 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/dream_sizes-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dream_sizes-1">
Past versions of dream_sizes-1.png
</button>
</p>
<div id="fig-dream_sizes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/dream_sizes-1.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Okay so for the smaller gene sets <code>lasso</code> and <code>elastic.net</code> are not really more sparse than Fisher’s exact test. But also, in these smaller gene sets FET</p>
<p>logistic SuSiE, with few exceptions return a single gene set with PIP <span class="math inline">\(&gt;0.95\)</span></p>
<pre class="r"><code>result %&gt;% ungroup() %&gt;%
  unnest(c(positives, thresh)) %&gt;%
  filter(near(thresh, t), GS_ID == &#39;go_bp&#39;) %&gt;%
  pivot_wider(
    id_cols = c(GS_ID, module.name), names_from=method, values_from = positives) %&gt;%
  select(!c(GS_ID, module.name)) %&gt;%
    GGally::ggpairs()</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/unnamed-chunk-3-1.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>result %&gt;% ungroup() %&gt;%
  unnest(c(positives, thresh)) %&gt;%
  filter(near(thresh, t), GS_ID == &#39;go_bp&#39;) %&gt;%
  pivot_wider(
    id_cols = c(GS_ID, module.name), names_from=method, values_from = positives) %&gt;%
  select(!c(GS_ID, module.name)) %&gt;%
    GGally::ggpairs()</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/unnamed-chunk-3-2.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>result %&gt;% ungroup() %&gt;%
  unnest(c(positives, thresh)) %&gt;%
  filter(near(thresh, t), GS_ID == &#39;go_bp&#39;) %&gt;%
  pivot_wider(
    id_cols = c(GS_ID, module.name), names_from=method, values_from = positives) %&gt;%
  select(!c(GS_ID, module.name)) %&gt;%
    GGally::ggpairs()</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-3">
Past versions of unnamed-chunk-3-3.png
</button>
</p>
<div id="fig-unnamed-chunk-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/unnamed-chunk-3-3.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="credible-sets" class="section level3">
<h3>Credible Sets</h3>
<pre class="r"><code>pattern &lt;- &#39;^dream.modules.count.cs_([^_])*$&#39;
result &lt;- tar_agg(pattern)

result %&gt;% filter(method %in% c(&#39;susie.veb.boost&#39;, &#39;logistic.susie.veb.boost&#39;)) %&gt;%
  select(method, module.name, GS_ID, cs_size) %&gt;% unnest(cs_size) %&gt;%
  filter(cs_size &lt; 100) %&gt;%
  ggplot(aes(x=cs_size, color=method)) + geom_histogram() + facet_wrap(vars(GS_ID)) + theme(aspect.ratio = 1)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/unnamed-chunk-4-1.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="number-of-small-credible-sets-per-gene-list" class="section level3">
<h3>Number of small credible sets per gene list</h3>
<p>Count the number of credible sets with &lt; 10 gene sets in each gene list</p>
<pre class="r"><code>result %&gt;% filter(method %in% c(&#39;susie.veb.boost&#39;, &#39;logistic.susie.veb.boost&#39;)) %&gt;%
  select(method, module.name, GS_ID, cs_size) %&gt;% unnest(cs_size) %&gt;%
  mutate(small_cset = cs_size &lt; 20) %&gt;%
  group_by(method, module.name, GS_ID) %&gt;%
  summarise(n_csets = sum(small_cset)) %&gt;%
  ggplot(aes(x=n_csets, color=method)) + geom_histogram() + facet_wrap(vars(GS_ID)) + theme(aspect.ratio = 1)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;method&#39;, &#39;module.name&#39;. You can override using the `.groups` argument.</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/gsea_comparisons.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/a3c53f36b3e70e77347843f59fed3c2f1a4c024e/docs/figure/gsea_comparisons.Rmd/unnamed-chunk-5-1.png" target="_blank">a3c53f3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.5        purrr_0.3.4       
 [5] readr_1.4.0        tidyr_1.1.3        tibble_3.1.1       ggplot2_3.3.3     
 [9] tidyverse_1.3.1    conflicted_1.0.4   targets_0.4.2.9000 workflowr_1.6.2   

loaded via a namespace (and not attached):
 [1] httr_1.4.2         sass_0.3.1         bit64_4.0.5        jsonlite_1.7.2    
 [5] modelr_0.1.8       bslib_0.2.4        assertthat_0.2.1   highr_0.9         
 [9] cellranger_1.1.0   yaml_2.2.1         pillar_1.6.0       backports_1.2.1   
[13] glue_1.4.2         digest_0.6.27      RColorBrewer_1.1-2 promises_1.2.0.1  
[17] rvest_1.0.0        colorspace_2.0-0   plyr_1.8.6         htmltools_0.5.1.1 
[21] httpuv_1.6.0       pkgconfig_2.0.3    broom_0.7.6        haven_2.4.1       
[25] scales_1.1.1       processx_3.5.1     whisker_0.4        later_1.2.0       
[29] git2r_0.28.0       generics_0.1.0     farver_2.1.0       ellipsis_0.3.1    
[33] cachem_1.0.4       withr_2.4.2        cli_2.5.0          magrittr_2.0.1    
[37] crayon_1.4.1       readxl_1.3.1       memoise_2.0.0      evaluate_0.14     
[41] ps_1.6.0           GGally_2.1.1.9000  fs_1.5.0           fansi_0.4.2       
[45] xml2_1.3.2         tools_4.0.5        data.table_1.14.0  hms_1.0.0         
[49] lifecycle_1.0.0    munsell_0.5.0      reprex_2.0.0       callr_3.7.0       
[53] compiler_4.0.5     jquerylib_0.1.4    rlang_0.4.10       grid_4.0.5        
[57] rstudioapi_0.13    igraph_1.2.6       labeling_0.4.2     rmarkdown_2.7     
[61] gtable_0.3.0       codetools_0.2-18   reshape_0.8.8      DBI_1.1.1         
[65] R6_2.5.0           lubridate_1.7.10   knitr_1.33         fastmap_1.1.0     
[69] bit_4.0.4          utf8_1.2.1         rprojroot_2.0.2    stringi_1.5.3     
[73] Rcpp_1.0.6         vctrs_0.3.7        dbplyr_2.1.1       tidyselect_1.1.0  
[77] xfun_0.22         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
