<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="karltayeb" />

<meta name="date" content="2021-04-26" />

<title>GSEA Method comparisons</title>

<script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">logistic_susie_gsea</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GSEA Method comparisons</h1>
<h4 class="author">karltayeb</h4>
<h4 class="date">2021-04-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>logistic_susie_gsea/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210205code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210205code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210205)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatreeff99bb3474ce325a3debe800dcd22d166e2b70cctargetblankff99bb3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karltayeb/logistic_susie_gsea/tree/ff99bb3474ce325a3debe800dcd22d166e2b70cc" target="_blank">ff99bb3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatreeff99bb3474ce325a3debe800dcd22d166e2b70cctargetblankff99bb3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karltayeb/logistic_susie_gsea/tree/ff99bb3474ce325a3debe800dcd22d166e2b70cc" target="_blank">ff99bb3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/GO_simulation_cache/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  GO_simulation_cache/
    Untracked:  _targets.R
    Untracked:  _targets/
    Untracked:  _targets2.R
    Untracked:  analysis/cache/
    Untracked:  analysis/initial_exploration.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  analysis/webgestaltr_list.Rmd
    Untracked:  cache/
    Untracked:  code/dream.targets.R
    Untracked:  code/fit.R
    Untracked:  code/gseabenchmark.targets.R
    Untracked:  code/logistic_susie.R
    Untracked:  code/plots.R
    Untracked:  code/score.R
    Untracked:  code/simulations.R
    Untracked:  code/utils.R
    Untracked:  code/webgestalt_example.R
    Untracked:  data/DREAM_modules/
    Untracked:  data/GO_Membership.RData
    Untracked:  data/WebGestalt/
    Untracked:  data/WebGestalt_cache/
    Untracked:  data/gene_list.txt
    Untracked:  data/webgestalt_example_genelist.txt
    Untracked:  output/WebGestalt/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   analysis/GO_simulation.Rmd
    Deleted:    analysis/gsea_comparisons.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/gseabenchmarkR_relevance.Rmd</code>) and HTML (<code>docs/gseabenchmarkR_relevance.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic_susie_gsea/blob/ff99bb3474ce325a3debe800dcd22d166e2b70cc/analysis/gseabenchmarkR_relevance.Rmd" target="_blank">ff99bb3</a>
</td>
<td>
karltayeb
</td>
<td>
2021-05-19
</td>
<td>
wflow_publish(“analysis/gseabenchmarkR_relevance.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(targets)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✓ ggplot2 3.3.3     ✓ purrr   0.3.4
✓ tibble  3.1.1     ✓ dplyr   1.0.6
✓ tidyr   1.1.3     ✓ stringr 1.4.0
✓ readr   1.4.0     ✓ forcats 0.5.1</code></pre>
<pre class="r"><code># load and row_bind all targets matching a pattern
tar_agg &lt;- function(pattern, selector=dplyr::matches){
  load.env &lt;- new.env()
  tar_load(selector(pattern), envir = load.env)
  result &lt;- grep(pattern , names(load.env), value=TRUE)
  result &lt;- do.call(&quot;list&quot;, mget(result, envir = load.env))
  result &lt;- bind_rows(result)
}</code></pre>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p><code>GSEABenchmarkR</code> provides a “relevance score” functionality. For a bunch of microarray and RNA seq experiments they curate a list of gene set rankings. These rankings can be compared against a gene set ranking output from your favorite GSEA method. Good GSEA methods will have rankings that agree with the curated rankings. Additionally, we want good GSEA methods to report relevant gene sets at reasonable thresholds.</p>
<p><code>GSEABenchmarkR</code> provides uniformly processed expression data from TCGA, and for each disease they provide “relevance scores” for gene sets in GO-BP, GO-MF, and KEGG. These relevance scores were derived from MalaCards. I’m not sure exactly how they were generated but they are supposed to serve as Silver standard to benchmark performance of GSEA methods.</p>
<p>Although in the paper they analyze 33 RNASeq datasets from TCGA, only there <code>GSEABenchmarkeR::loadEData("tcga")</code> only returns 15 data sets. We follow a vanilla DE pipeline outlined in the <code>GSEABenchmarkR</code> documentation, and then apply a suite of gene list based GSEA methods. Finally we use the MalaCards derived relevance scores to evaluate results.</p>
<pre class="r"><code>result &lt;- tar_agg(&#39;gseabenchmark.score&#39;, dplyr::starts_with)

result %&gt;% mutate(rel = relevance/opt.relevance) %&gt;%
  filter(near(thresh, 0)) %&gt;%
  ggplot(aes(x=method, y = rel, fill=method)) + 
  geom_boxplot() + facet_wrap(vars(n_genes)) +
  theme(aspect=1.0, axis.text.x=element_blank())</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We evaluate the relevance of elastic net, ORA, and logistic SuSiE when we use gene lists comprised of the top 100, 500, or 1000 DE genes for each of the 15 cancer datasets. Logistic SuSiE does better than the others, which indicates that the gene set ranking given by PIPs does a better job prioritizing relevant gene sets.</p>
<pre class="r"><code>result %&gt;% mutate(rel = relevance/opt.relevance) %&gt;%
  filter(thresh %in% c(0, 0.1, 0.9, 0.95)) %&gt;%
  ggplot(aes(x=method, y = rel, fill=method)) + 
  geom_boxplot() + facet_grid(vars(thresh), vars(n_genes)) +
  theme(aspect=1.0, axis.text.x=element_blank())</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Unfortunately, this does not mean that the relevant gene sets have particularly large PIPs. Here we threshold on PIP (or 1 - p for FET), we keep the top ranking gene-sets above the threshold. All gene sets below the threshold are given equal rank. The relevance score for logistic SuSiE drops off even with a modest threshold of 0.1. That is, none of the gene sets with PIPs &gt; 0.1 are even listed among the relevant gene sets.</p>
<p>As we’ll see most of the components have very small credible sets, so the PIP threshold of 0.1 is not that stringent. Rather, this is telling us that the relevant gene sets have small but slightly elevated PIPs that we would not necessarily pick.</p>
<pre class="r"><code>result &lt;- tar_agg(&#39;gseabenchmark.cs&#39;, dplyr::starts_with)

r2 &lt;- result %&gt;% filter(near(target_coverage, 0.95)) %&gt;%
  select(method, n_genes, ID, cs_size) %&gt;% unnest(cs_size) %&gt;%
  mutate(cs.bin = cut(cs_size, c(0, 1, 10, 100, 1000, 10000), labels = c(1, 10, 100, 1000, 10000)))

r2 %&gt;% ggplot(aes(x=cs.bin)) + geom_bar() + facet_wrap(vars(n_genes)) + theme(aspect=1.0)</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/cset_size_95-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>r2$cs.bin[1]</code></pre>
<pre><code>[1] 10000
Levels: 1 10 100 1000 10000</code></pre>
<pre class="r"><code>r2 %&gt;% mutate(small.cs = (cs.bin %in% c(1, 10, 100))) %&gt;% group_by(method, ID, n_genes) %&gt;% 
  summarise(n.small.cs = sum(small.cs))%&gt;%
  ggplot(aes(x=n.small.cs)) + geom_histogram() + facet_wrap(vars(n_genes)) + theme(aspect=1.0)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;method&#39;, &#39;ID&#39;. You can override using the `.groups` argument.</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/cset_size_95-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Defining small credible sets as 95% credible sets with fewer than 100 gene sets, we see that many for many cancer types, logistic SuSiE reports 0 small credible sets.</p>
<p>The situation is not improved by considering credible sets with lower coverage. For example, we can look at the 10% credible sets.</p>
<pre class="r"><code>r2 &lt;- result %&gt;% filter(near(target_coverage, 0.1)) %&gt;%
  select(method, n_genes, ID, cs_size) %&gt;% unnest(cs_size) %&gt;%
  mutate(cs.bin = cut(cs_size, c(0, 1, 10, 100, 1000, 10000), labels = c(1, 10, 100, 1000, 10000)))

r2 %&gt;% ggplot(aes(x=cs.bin)) + geom_bar() + facet_wrap(vars(n_genes)) + theme(aspect=1.0)</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/cset_sizes_10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>r2 %&gt;% mutate(small.cs = (cs.bin %in% c(1, 10, 100))) %&gt;% group_by(method, ID, n_genes) %&gt;% 
  summarise(n.small.cs = sum(small.cs))%&gt;%
  ggplot(aes(x=n.small.cs)) + geom_histogram() + facet_wrap(vars(n_genes)) + theme(aspect=1.0)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;method&#39;, &#39;ID&#39;. You can override using the `.groups` argument.</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/cset_sizes_10-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="are-the-hits-related-to-the-high-relevance-gene-sets" class="section level3">
<h3>Are the hits related to the high relevance gene sets?</h3>
<p>We’ve seen that logistic SuSiE is returning a few small credible sets, mostly singletons (ie a single gene sets with PIP close to 1). We’ve also seen that these gene sets are NOT the “high relevance” gene sets provided by <code>GSEABenchmarkR</code>, but that these gene sets do show slightly elevated, but small PIPs.</p>
<p>Are the gene sets that logistic SuSie picks out related to the relevant gene sets?</p>
<pre class="r"><code>tar_load(mala)
tar_load(gseabenchmark.fit_logistic.susie.veb.boost32_1000)
tar_load(gseabenchmark.fit_fet_1000)
tar_load(gseabenchmark.fit_elastic.net_1000)

tar_load(X.eb.gobp)

overlap &lt;- Matrix::t(X.eb.gobp) %*% X.eb.gobp
gs.size &lt;- setNames(as.vector(rep(1, dim(X.eb.gobp)[1]) %*% X.eb.gobp), colnames(X.eb.gobp))
union &lt;- outer(gs.size, gs.size, &#39;+&#39;) - overlap
jaccard &lt;- overlap / union
asymmetric &lt;- apply(overlap, 1, function(x) x / gs.size)</code></pre>
<pre class="r"><code>library(S4Vectors)</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre class="r"><code>par(mfrow=c(2, 2))
gobp.relevance &lt;- mala$mala.gobp
logistic.susie.pip &lt;- gseabenchmark.fit_logistic.susie.veb.boost32_1000 %&gt;% select(ID, y, pip)

pip.sets &lt;- names(which(logistic.susie.pip$pip[[1]] &gt; 0.95))
relevant.sets &lt;- intersect(colnames(jaccard), rownames(gobp.relevance$BLCA))

genes &lt;- names(which(logistic.susie.pip$y[[1]][, 1]) != 0)

hist(
  as.vector(X.eb.gobp[genes, relevant.sets] %*% rep(1, length(relevant.sets))),
  main = &#39;relevant gene sets per gene in list&#39;
)

hist(
  as.vector(X.eb.gobp[genes, pip.sets] %*% rep(1, length(pip.sets))),
  main = &#39;pip gene sets per gene in list&#39;
)

image(t(as.matrix(jaccard[pip.sets, relevant.sets])), keep.dendro = F, main=&#39;Jaccard Index&#39;, ylab = &#39;large pip gene sets&#39;, xlab = &#39;relevant gene sets&#39;, )</code></pre>
<pre><code>Warning in plot.window(...): &quot;keep.dendro&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in plot.xy(xy, type, ...): &quot;keep.dendro&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in axis(side = side, at = at, labels = labels, ...): &quot;keep.dendro&quot; is
not a graphical parameter

Warning in axis(side = side, at = at, labels = labels, ...): &quot;keep.dendro&quot; is
not a graphical parameter</code></pre>
<pre><code>Warning in box(...): &quot;keep.dendro&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in title(...): &quot;keep.dendro&quot; is not a graphical parameter</code></pre>
<pre class="r"><code>boxplot(
  as.vector(rep(1, length(genes)) %*% X.eb.gobp[genes, relevant.sets] / gs.size[relevant.sets]),
  as.vector(rep(1, length(genes)) %*% X.eb.gobp[genes, pip.sets] / gs.size[pip.sets]),
  main = &#39;proportion of gene set in gene list&#39;, names=c(&#39;relevant&#39;, &#39;pip&#39;)
)</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/BLCA-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The gene sets with large pips have comparatively large overlap with the gene list compared to relevant gene sets. We notice that many of the DE genes participate in more than one gene set. Logistic SuSiE picked out gene sets that largely partition the gene list. That is, each gene in the list is represented by at most one gene set with large pip.</p>
<p>You might think that SuSiE is picking more specific gene sets, surprisingly that is not the case. Below, we see that the gene sets with large PIP are larger than the relevant gene sets.</p>
<pre class="r"><code>logistic.susie.pip &lt;- logistic.susie.pip %&gt;% rowwise() %&gt;%
  mutate(
    pip.sets = list(names(which(pip &gt; 0.95))),
    relevant.sets = list(intersect(colnames(jaccard), rownames(gobp.relevance[[ID]]))),
    genes = list(names(which(y[, 1]) != 0)),
    relevant.set.overlap = list(as.vector(rep(1, length(genes)) %*% X.eb.gobp[genes, relevant.sets])),
    relevant.set.size = list(as.vector(gs.size[relevant.sets])),
    pip.set.overlap = list(as.vector(rep(1, length(genes)) %*% X.eb.gobp[genes, pip.sets])),
    pip.set.size = list(as.vector(gs.size[pip.sets])),
    total.set.overlap = list(as.vector(rep(1, length(genes)) %*% X.eb.gobp[genes,])),
    total.set.size = list(as.vector(gs.size))
)

fet.result &lt;- gseabenchmark.fit_fet_1000 %&gt;% rowwise() %&gt;%
  mutate(
    pip.sets = list(names(which(one_minus_bh_p &gt; 0.95))),
    pip.set.overlap = list(as.vector(rep(1, length(genes)) %*% X.eb.gobp[genes, pip.sets])),
    pip.set.size = list(as.vector(gs.size[pip.sets])),
  )

en.result &lt;- gseabenchmark.fit_elastic.net_1000 %&gt;% rowwise() %&gt;%
  mutate(
    pip.sets = list(names(which(selected &gt; 0.95))),
    pip.set.overlap = list(as.vector(rep(1, length(genes)) %*% X.eb.gobp[genes, pip.sets])),
    pip.set.size = list(as.vector(gs.size[pip.sets])),
  )

result &lt;- bind_rows(
  logistic.susie.pip %&gt;%
  select(ID, relevant.set.overlap, relevant.set.size) %&gt;% 
  unnest(-ID) %&gt;% 
  mutate(
    size = relevant.set.size,
    prop.overlap = relevant.set.overlap / relevant.set.size,
    gene.set.type = &#39;relevant&#39;
  ) %&gt;% select(ID, gene.set.type, prop.overlap, size),
  logistic.susie.pip %&gt;%
    select(ID, pip.set.overlap, pip.set.size) %&gt;% 
    unnest(-ID) %&gt;% 
    mutate(
      size = pip.set.size,
      prop.overlap = pip.set.overlap / pip.set.size,
      gene.set.type = &#39;logistic.susie&#39;
    ) %&gt;% select(ID, gene.set.type, prop.overlap, size),
  logistic.susie.pip %&gt;%
  select(ID, total.set.overlap, total.set.size) %&gt;% 
  unnest(-ID) %&gt;% 
  mutate(
    size = total.set.size,
    prop.overlap = total.set.overlap / total.set.size,
    gene.set.type = &#39;all&#39;
  ) %&gt;% select(ID, gene.set.type, prop.overlap, size),
  fet.result %&gt;%
  select(ID, pip.set.overlap, pip.set.size) %&gt;% 
  unnest(-ID) %&gt;% 
  mutate(
    size = pip.set.size,
    prop.overlap = pip.set.overlap / pip.set.size,
    gene.set.type = &#39;fet&#39;
  ) %&gt;% select(ID, gene.set.type, prop.overlap, size),
  en.result %&gt;%
  select(ID, pip.set.overlap, pip.set.size) %&gt;% 
  unnest(-ID) %&gt;% 
  mutate(
    size = pip.set.size,
    prop.overlap = pip.set.overlap / pip.set.size,
    gene.set.type = &#39;elastic.net&#39;
  ) %&gt;% select(ID, gene.set.type, prop.overlap, size)
)


result %&gt;% ggplot(aes(x=gene.set.type, y = prop.overlap)) + geom_boxplot()</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/size_overlap-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>result %&gt;% ggplot(aes(x=gene.set.type, y = size)) + geom_boxplot()</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/size_overlap-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot() +
  geom_point(data=filter(result, gene.set.type != &#39;all&#39;),
             aes(x=prop.overlap, y = size, color=gene.set.type),
             position = &#39;dodge&#39;)</code></pre>
<pre><code>Warning: Width not defined. Set with `position_dodge(width = ?)`</code></pre>
<p><img src="figure/gseabenchmarkR_relevance.Rmd/size_overlap-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>The gene sets SuSiE is picking out have a higher proportion overlap with the gene list than the “relevant” gene sets. Can’t blame the method for doing that!</p>
<p>We also see that the size of the credible sets SuSiE is picking are quite a bit larger on average than the relevant gene sets. Competing gene sets against each other is really effective when the gene list does not contain many false positives– and should pick out the most narrow gene set. It is more challenging to pick out the most specific gene set when there is substantial noise, however.</p>
<pre class="r"><code>knitr::knit_exit()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
